### PR TITLE
PIMOB-1427: Fix card number input should be blue when I'm filling it in

### PR DESCRIPTION
**Pull request by ehab-al-cko**
[Jira ticket](https://checkout.atlassian.net/browse/PIMOB-1427?atlOrigin=eyJpIjoiY2VhZDExOTFmYjhkNDg3NWJhNDg2Zjg2ZDg0MWMwMTkiLCJwIjoiaiJ9)

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/UI/CommonUI/View/Component/InputView.swift [hunk 1]
```diff
@@ -174,7 +174,9 @@ extension InputView {
 
 extension InputView: TextFieldViewDelegate {
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        delegate?.textField(textField, shouldChangeCharactersIn: range, replacementString: string) ?? true
+        let shouldChangeCharacter = delegate?.textField(textField, shouldChangeCharactersIn: range, replacementString: string) ?? true
+        textFieldContainerBorder.layer.borderColor = style?.textfield.focusBorderColor.cgColor
+        return shouldChangeCharacter
```
Comment (alex-ioja-yang-cko) [inline Source/UI/CommonUI/View/Component/InputView.swift:179]: Is this callback only getting triggered as textfield input changes? Shouldn't we instead use `shouldBeginEditing` to set the focus color?

File: iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift [hunk 2]
```diff
@@ -17,8 +17,11 @@ final class FrameUITests: XCTestCase {
         app.launch()
 
         app.tapButton(name: "UITestDefault")
-        let cardNumberTextField = app.scrollViews.children(matching: .other).element(boundBy: 0).children(matching: .other).element(boundBy: 2).children(matching: .other).element.children(matching: .other).element.children(matching: .other).element(boundBy: 1).children(matching: .other).element(boundBy: 1).children(matching: .other).element
-        let expiryTextField = app.scrollViews.children(matching: .other).element(boundBy: 0).children(matching: .other).element(boundBy: 3).children(matching: .other).element.children(matching: .other).element.children(matching: .other).element(boundBy: 1).children(matching: .other).element.children(matching: .other).element
+
+        // MARK: Full UI Payment
```
Comment (alex-ioja-yang-cko) [inline iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift:21]: I don't think this Mark is needed here, appears to have been copied from separate test that was more complex?

File: Source/UI/CommonUI/View/Component/InputView.swift [hunk 3]
```diff
@@ -168,12 +174,14 @@ extension InputView {
...
```
Comment (alex-ioja-yang-cko) [inline Source/UI/CommonUI/View/Component/InputView.swift:179]:
```suggestion
        textFieldContainerBorder.layer.borderColor = style?.textfield.focusBorderColor.cgColor
        return delegate?.textField(textField, shouldChangeCharactersIn: range, replacementString: string) ?? true
```

**Review by alex-ioja-yang-cko — COMMENTED** [review]
Looks good, just small suggestion to group the code logically, which then lead to removing the property declaration altogether...

Will approve once main is updated from other PR.

File: iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift (hunk 2 above)
Comment (ehab-al-cko) [reply iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift:21]: Yes, i will removed

File: Source/UI/CommonUI/View/Component/InputView.swift (hunk 3 above)
Comment (ehab-al-cko) [reply Source/UI/CommonUI/View/Component/InputView.swift:179]: I have tried this suggest and color still gray, 
`textFieldContainerBorder.layer.borderColor = style?.textfield.focusBorderColor.cgColor` needs to be set after `delegate?.textField(textField, shouldChangeCharactersIn: range, replacementString: string)` logic is finished

Comment (alex-ioja-yang-cko) [reply Source/UI/CommonUI/View/Component/InputView.swift:179]: My bad, thanks for explaining this 👍

File: iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift (hunk 2 above)
Comment (ehab-al-cko) [inline iOS Example Frame/iOS Example FrameUITests/FrameUITests.swift:21]:
```suggestion
```

**Review by alex-ioja-yang-cko — APPROVED** [review]

**Pull request merged**